### PR TITLE
fix: Fix most 'path' bugs. Also adds some logging

### PR DESCRIPTION
**Pull request by lemire**
Up until now, we had almost no testing for paths when parsing URLs. Thankfully, it seems that we only had many shallow bugs. I hope that I fixed most of them with this PR.

I also added basic logging. The approach I took to logging is very crude: set a macro, and you get lots of printouts. That's not the best approach but it is better than no logging at all. The proper way to do logging for a library like ada would be to do like simdjson (https://github.com/simdjson/simdjson/discussions/1938). Instead of ugly macros all over the place, you have have an actual design. However, this would take too much time at this stage and we have slightly more urgent matters.

Fixes: https://github.com/ada-url/ada/issues/113
Fixes: https://github.com/ada-url/ada/issues/86

### CODE REVIEW

**Comment by lemire** [comment]
I am going to see what we can do to make windows more sane.

File: include/ada/checkers.h [hunk 1]
```diff
@@ -59,11 +59,17 @@ namespace ada::checkers {
...
   /**
-   * @details A Windows drive letter is two code points, of which the first is an ASCII alpha
-   * and the second is either U+003A (:) or U+007C (|).
+   * @details A string starts with a Windows drive letter if all of the following are true:
+   *
+   *   - its length is greater than or equal to 2
+   *   - its first two code points are a Windows drive letter
+   *   - its length is 2 or its third code point is U+002F (/), U+005C (\), U+003F (?), or U+0023 (#).
+   *
+   * https://url.spec.whatwg.org/#start-with-a-windows-drive-letter
```
Comment (anonrig) [inline include/ada/checkers.h:68]: This should be starting with “@see”

File: src/url.cpp [hunk 2]
```diff
@@ -44,7 +47,11 @@ namespace ada {
 
 
   ada_really_inline bool url::parse_prepared_path(std::string_view input) {
-    path.clear();
+#if ADA_LOGGING
+    std::cout << "url::parse_prepared_path(" << input << ")" << std::endl;
+    std::cout << "url.path = " << path << std::endl;
+#endif
+    //path.clear();
```
[the remark this answers is not in the document]
Comment (lemire) [reply src/url.cpp:54]: Solved.

File: include/ada/checkers.h (hunk 1 above)
Comment (lemire) [reply include/ada/checkers.h:68]: Solved.

**Comment by lemire** [comment]
Merging. We will have to do more work to get this code to avoid undefined behavior.

**Pull request merged**